pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -342,3 +283,62 @@ func (p *AgentPoller) Stop() {
 		p.agentDone = nil
 	}
 }
+
+// With libvirt 5.6.0 direct call to agent can be replaced with call to libvirt Domain.GetGuestInfo
+func executeAgentCommands(commands []AgentCommand, con cli.Connection, agentStore *AsyncAgentStore, domainName string) {
+	for _, command := range commands {
+		// replace with direct call to libvirt function when 5.6.0 is available
+		cmdResult, err := con.QemuAgentCommand(`{"execute":"`+string(command)+`"}`, domainName)
+		if err != nil {
+			// skip the command on error, it is not vital
+			continue
+		}
+
+		// parse the json data and convert to domain api
+		// for libvirt 5.6.0 json conversion deprecated
+		switch command {
+		case GET_INTERFACES:
+			interfaces, err := parseInterfaces(cmdResult)
+			if err != nil {
+				log.Log.Errorf("Cannot parse guest agent interface %s", err.Error())
+			}
+			agentStore.Store(GET_INTERFACES, interfaces)
```

Comment:
This block repeats a lot, consider please to add a function with parameterize

Reply:
I have chosen not to touch this part of the code and focus only on splitting the functionality as described in the commit message.